File: examples/governance/compliance_domain/lib/compliance_domain/regulatory_framework/events/activated_framework.rb
module ComplianceDomain
  class RegulatoryFramework
    module Events
      class ActivatedFramework
        attr_reader :aggregate_id, :framework_id, :effective_date, :name, :jurisdiction, :version, :authority, :requirements, :status, :occurred_at

        def initialize(aggregate_id: nil, framework_id: nil, effective_date: nil, name: nil, jurisdiction: nil, version: nil, authority: nil, requirements: nil, status: nil)
          @aggregate_id = aggregate_id
          @framework_id = framework_id
          @effective_date = effective_date
          @name = name
          @jurisdiction = jurisdiction
          @version = version
          @authority = authority
          @requirements = requirements
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end

File: examples/governance_static_go/domain/activated_framework.go
package domain

import "time"

type ActivatedFramework struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	FrameworkId string `json:"framework_id"`
	EffectiveDate time.Time `json:"effective_date"`
	Name string `json:"name"`
	Jurisdiction string `json:"jurisdiction"`
	Version string `json:"version"`
	Authority string `json:"authority"`
	Requirements []FrameworkRequirement `json:"requirements"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e ActivatedFramework) EventName() string { return "ActivatedFramework" }

func (e ActivatedFramework) GetOccurredAt() time.Time { return e.OccurredAt }
